BHoM_Adapter/HelperMethods/CopyBHoMObjectProperties.cs

```diff
@@ -32,13 +32,15 @@
 using System.Linq;
 using System.Reflection;
 
-namespace BH.Engine.Adapter
+namespace BH.Adapter
 {
-    public static partial class Modify
+    public abstract partial class BHoMAdapter
     {
         [Description("Gets called during the Push. Takes properties specified from the source IBHoMObject and assigns them to the target IBHoMObject.")]
-        public static void CopyBHoMObjectProperties<T>(T target, T source, string AdapterIdName) where T : class, IBHoMObject
+        public void CopyBHoMObjectProperties<T>(T target, T source, string adapterIdName) where T : class, IBHoMObject
         {
+            target = target.DeepClone();
```

Comment:
Dont think we want to do this clone here. Afraid this could break the full dependency chain.

Reply:
Thanks! Sorry this was a leftover from a previous solution